ui/components/svg.vue

```diff
@@ -76,59 +73,43 @@ export default {
           colors = [colorLookup[this.name]];
         }
 
-        return getColorCircle(colors, this.name);
+        return getColorCircle(colors, this.title || this.name);
       }
 
-      return getSvg(this.name);
+      return getSvg(this.name, this.type, this.title);
     }
   }
 };
 
 
 /**
  * Returns the contents of the provided SVG file as an inline SVG.
- * @param {string} svgBasename Name of the file (withoug extension).
- * @param {array.<string>} classNames List of class names the <svg> tag should have.
+ * @param {string} name Name of the icon (without extension).
+ * @param {string|null} category The category (directory) of the icon.
+ * @param {string|null} title An optional (tooltip) title for the icon.
  * @returns {string} The inline <svg> tag or an empty string if the file was not found.
  */
-function getSvg(svgBasename, classNames = []) {
-  let svg = svgBasename in icons ? icons[svgBasename].replace(/\s+$/, ``) : ``;
-
-  if (classNames.length > 0) {
-    svg = svg.replace(/<svg([^>]*)>/, `<svg$1 class="${classNames.join(` `)}">`);
+function getSvg(name, category = null, title) {
+  if (name === ``) {
+    return ``;
   }
 
-  return svg;
-}
-
-
-/**
- * Get an icon for the provided category.
- * @param {string} categoryName Name of the category.
- * @param {array.<string>} classNames List of class names the <svg> tag should have.
- * @returns {string} The inline <svg> tag or an empty string if the file was not found.
- */
-function getCategoryIcon(categoryName, classNames = []) {
-  const sanitzedCategoryName = categoryName.toLowerCase().replace(/[^\w]+/g, `-`);
-  classNames.push(`category-${sanitzedCategoryName}`, `category-icon`);
-
-  const svg = getSvg(`categories/${sanitzedCategoryName}`, classNames);
-  return svg.replace(/(<svg[^>]*>)/, `$1<title>Category: ${categoryName}</title>`);
-}
+  const kebabName = name.replace(/([a-z])([A-Z])/g, `$1-$2`).toLowerCase().replace(/[^\w]+/g, `-`);
+  const svgBasename = (category ? `${category}/` : ``) + kebabName;
```

Comment:
Or: `const svgBasename = path.join(category || ``, kebabName);` You can omit the ` || `` ` if you set the parameter default to an empty string.

Reply:
We can't use `path` because it's only available in Node environment, not in the browser.